lambda_src/lambda_function.py

```diff
@@ -155,30 +220,80 @@ def sync_flow(event: Any, context: Any = None) -> Dict[Text, Any]:
             'statusCode': 200,
             'body': b64encode(compress(data_dumps.encode())).decode(),
             'isBase64Encoded': True,
-            'headers': {'Content-Encoding': 'gzip'}
+            'headers': {'Content-Encoding': 'gzip'},
         }
+        end_time = timer()
+        if response and (end_time - start_time) > 20:
+            LOG.debug('Storing the response in lock cache.')
+            try:
+                finish_batch_processing(batch_id, response)  # write the response
+            except ClientError as ce:
+                if ce.response['Error']['Code'] == 'ValidationException':
+                    LOG.error(ce)
+                    error_dumps = dumps(
+                        {
+                            'data': [
+                                [
+                                    rn,
+                                    {
+                                        'error': f'Response size ({len(dumps(response))} bytes) too large to be stored in the backend.',
+                                        'response_hash': md5(
+                                            dumps(response, sort_keys=True).encode()
+                                        ).hexdigest(),
+                                    },
+                                ]
+                                for rn, *args in req_body['data']
```

Comment:
maybe `res_data` instead of `req_body['data']` and then you can reconstruct the response sizes?

Reply:
Something like below maybe? Would it be fine, or does it seem cluttered?
![Screenshot 2023-01-06 at 10 26 39 PM](https://user-images.githubusercontent.com/77716642/211059938-647010e5-bf7f-4f48-8a02-3dfe3325d61c.png)